### PR TITLE
feat(mutation testing elements): Implement new mutation testing elements html reporter

### PR DESCRIPTION
**Pull request by nicojs**

### CODE REVIEW

**Comment by nicojs** [comment]
I think we should "Rebase and merge" rather than "Squash and merge", better for the changelog.

File: packages/core/src/reporters/MutationTestReportCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  private mutationTestReport(results: ReadonlyArray<MutantResult>): mutationTestReportSchema.MutationTestResult {
+    return {
+      files: this.toFileResults(results),
+      schemaVersion: '1.0',
+      thresholds: this.options.thresholds
+    };
+  }
+
+  private toFileResults(results: ReadonlyArray<MutantResult>): mutationTestReportSchema.FileResultDictionary {
+    const files: mutationTestReportSchema.FileResultDictionary = Object.create(null);
```
Comment (simondel) [inline packages/core/src/reporters/MutationTestReportCalculator.ts]: This is quite sneaky. You're creating a variable with the same name as an instance property

File: packages/core/src/reporters/MutationTestReportCalculator.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    results.forEach(mutantResult => {
+      const fileResult = files[mutantResult.sourceFilePath];
+      if (fileResult) {
+        fileResult.mutants.push(this.toMutantResult(mutantResult));
+      } else {
+        const sourceFile = this.files.files.find(file => file.name === mutantResult.sourceFilePath);
```
Comment (simondel) [inline packages/core/src/reporters/MutationTestReportCalculator.ts]: You're inside of an anonymous function, referencing `this.files.files` while you have a variable in the method above called `files` and you have an instance variable called `files` as well. It's quite confusing which files is which and why you need to do `files.files`

File: packages/html-reporter/src/templates/customElements.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+interface MutationTestReportAppHTMLElement {
+  titlePostfix: string;
+}
+declare namespace JSX {
+  interface IntrinsicElements {
```
Comment (simondel) [inline packages/html-reporter/src/templates/customElements.ts]: Where is this used?

Comment (nicojs) [reply packages/html-reporter/src/templates/customElements.ts]: It is how to add custom elements to typed-html: https://www.npmjs.com/package/typed-html#custom-elements

**Comment by nicojs** [comment]
@simondel  I've fixed/commented on all review comments, is this better?

**Pull request merged**